### ISSUE TITLE
paddle.fluid.core.EnforceNotMet: Tensor holds the wrong type, it holds double at [/home/vis/fuyi/networkCode/Paddle/paddle/fluid/framework/tensor_impl.h:29]

### ISSUE THREAD

**Issue by fuyi02** [PaddlePaddle/Paddle#17806]
get error while use square_error_cost op，
I want to calculate a loss like mseloss in pytorch ,I try to use square_error_cost and sqrt op to make it work, but I get error like this:

my code:
def get_mse_loss(logit, label, ignore_mask):
    shape = logit.shape
    logit = fluid.layers.sigmoid(logit) 
    square_loss = fluid.layers.square_error_cost(logit, label)                                                                                                              
    loss = fluid.layers.sqrt(square_loss)
    loss = loss * ignore_mask
    avg_loss = fluid.layers.reduce_sum(loss) / fluid.layers.reduce_sum(ignore_mask)
    return avg_loss

error:
paddle.fluid.core.EnforceNotMet: Tensor holds the wrong type, it holds double at [/home/vis/fuyi/networkCode/Paddle/paddle/fluid/framework/tensor_impl.h:29
  loss, pred = train_exe.run(fetch_list, feed={'image': imgs, 'label': grts, 'mask': masks})
  File "/home/vis/lixubin/paddle/python276.withsem/lib/python2.7/site-packages/paddle/fluid/parallel_executor.py", line 297, in run
    self.executor.run(fetch_list, fetch_var_name)
paddle.fluid.core.EnforceNotMet: Tensor holds the wrong type, it holds double at [/home/vis/fuyi/networkCode/Paddle/paddle/fluid/framework/tensor_impl.h:29]
PaddlePaddle Call Stacks: 
0       0x7fb9a8ec8cd5p void paddle::platform::EnforceNotMet::Init<char const*>(char const*, char const*, int) + 357
1       0x7fb9a8ec9059p paddle::platform::EnforceNotMet::EnforceNotMet(std::__exception_ptr::exception_ptr, char const*, int) + 137
2       0x7fb9a8ecff41p float const* paddle::framework::Tensor::data<float>() const + 193
3       0x7fb9aa38782ap void paddle::operators::ElementwiseComputeEx<paddle::operators::SubFunctor<float>, paddle::platform::CUDADeviceContext, float, float>(paddle::framework::ExecutionContext const&, paddle::framework::Tensor const*, paddle::framework::Tensor const*, int, paddle::operators::SubFunctor<float>, paddle::framework::Tensor*) + 90
4       0x7fb9aa388ac9p paddle::operators::ElementwiseSubKernel<paddle::platform::CUDADeviceContext, float>::Compute(paddle::framework::ExecutionContext const&) const + 457
5       0x7fb9aa389073p std::_Function_handler<void (paddle::framework::ExecutionContext const&), paddle::framework::OpKernelRegistrarFunctor<paddle::platform::CUDAPlace, false, 0ul, paddle::operators::ElementwiseSubKernel<paddle::platform::CUDADeviceContext, float>, paddle::operators::ElementwiseSubKernel<paddle::platform::CUDADeviceContext, paddle::platform::float16>, paddle::operators::ElementwiseSubKernel<paddle::platform::CUDADeviceContext, double>, paddle::operators::ElementwiseSubKernel<paddle::platform::CUDADeviceContext, int>, paddle::operators::ElementwiseSubKernel<paddle::platform::CUDADeviceContext, long> >::operator()(char const*, char const*, int) const::{lambda(paddle::framework::ExecutionContext const&)#1}>::_M_invoke(std::_Any_data const&, paddle::framework::ExecutionContext const&) + 35
6       0x7fb9aa87b49cp paddle::framework::OperatorWithKernel::RunImpl(paddle::framework::Scope const&, boost::variant<paddle::platform::CUDAPlace, paddle::platform::CPUPlace, paddle::platform::CUDAPinnedPlace, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_> const&) const + 844
7       0x7fb9aa8796a4p paddle::framework::OperatorBase::Run(paddle::framework::Scope const&, boost::variant<paddle::platform::CUDAPlace, paddle::platform::CPUPlace, paddle::platform::CUDAPinnedPlace, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_> const&) + 340
8       0x7fb9aa71222ap paddle::framework::details::ComputationOpHandle::RunImpl() + 250
9       0x7fb9aa70b906p paddle::framework::details::OpHandleBase::Run(bool) + 118
10      0x7fb9aa6a8ffdp
11      0x7fb9aa041f13p std::_Function_handler<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> (), std::__future_base::_Task_setter<std::unique_ptr<std::__future_base::_Result<void>, std::__future_base::_Result_base::_Deleter>, void> >::_M_invoke(std::_Any_data const&) + 35
12      0x7fb9a9b294c7p std::__future_base::_State_base::_M_do_set(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>&, bool&) + 39
13      0x7fb9e27b0973p pthread_once + 83
14      0x7fb9aa6a7d12p
15      0x7fb9a9b2a8e4p ThreadPool::ThreadPool(unsigned long)::{lambda()#1}::operator()() const + 404
16      0x7fb9bbd268a0p
17      0x7fb9e27ab1c3p
18      0x7fb9e152412dp clone + 109

logit data type is float32
label data type is float32

I don't know what happens..  can anyone give me some advice?

**Comment by sandyhouse** [comment]
输入数据是什么格式，是否可以打印变量看一下具体的类型

**Comment by sandyhouse** [comment]
程序中虽然label的定义是float32的，但实际feed到label的值是float64的，导致运行时label的值为float64（double）。

**Comment by sandyhouse** [comment]
Closing this issue. Reopen it if you have any further question.
